smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HTTPBindingProtocolGenerator.kt

```diff
@@ -220,17 +223,40 @@ abstract class HTTPBindingProtocolGenerator(
                         // get all members sorted by name and filter out either all members with other traits OR members with the payload trait
                         val httpBodyMembers = members.filter { it.isInHttpBody() }
                         val path = "properties.".takeIf { shape.hasTrait<ErrorTrait>() } ?: ""
-                        writer.write("")
```

Comment:
Check if the `NeedsWriter` trait is present before generating a writer, and that the `NeedsReader` trait is present before generating a reader.

(code looks substantially different because it was reformatted but the method calls are identical.)